### PR TITLE
add Control view restrictions of the Environment Variable values

### LINKED ISSUE #2289: Control view restrictions of the Environment Variable values
### Description

> Requested by @biancode

@biancode runs multiple teams in FlowForge for their various customer projects. In each team, they invite their respective customers, and developer their flows, but provide the customers access to their configuration/flows, etc.

one thing that they do not want to be giving customers access to though are some of the values of their environment variables/secrets.

After talking it through, we came to the conclusion that the option of a Team Owner being able to control whether or not the Environment Variable values (on a var by var basis) can be seen by Team Members would be a good solution for this.

### Have you provided an initial effort estimate for this issue?

I have provided an initial effort estimate

### PR DESCRIPTION
**Pull request by sumitshinde-84**
## Description

Added feature to control view restrictions of Environment Variable values

- Implemented a new feature that allows administrators to control the view restrictions of Environment Variable values.
- If the user is an admin, they now have the ability to see and control the view restrictions of the Environment Variable values.
- The feature enhances the existing functionality by providing fine-grained control over who can view the values of    Environment Variables.
- This feature helps improve security and privacy by limiting access to sensitive environment information.

https://github.com/flowforge/flowforge/assets/110285294/30613382-8aaa-46ce-b0c5-7bf656f14da2

closes #2289

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/flowforge/flowforge/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `flowforge/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `flowforge/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Backport needed? -> add the `backport` label
 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 1]
```diff
@@ -25,22 +25,23 @@
                             :error="item.error"
                             :disabled="item.encrypted"
                             value-empty-text=""
-                            :type="(!readOnly && (editTemplate || item.policy === undefined))?'text':'uneditable'"></FormRow>
-                        <!-- <ff-text-input  v-model="item.name" :disabled="item.encrypted"  /> -->
+                            :type="(!readOnly && (editTemplate || item.policy === undefined)) ? 'text' : 'uneditable'"
+                        ></FormRow>
                     </td>
-                    <td class="px-4 py-4 border w-auto align-top" :class="{'align-middle':item.encrypted}">
-                        <div class="w-full" v-if="!item.encrypted">
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/sections/Environment.vue:32]: I believe we'll still want the `item.encrypted` feature to show the padlock icon. The `settings['team:environment-variable-view']` can be added as an extra fork the if statement

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 2]
```diff
@@ -25,22 +25,23 @@
...
+                    <td class="px-4 py-4 border w-auto align-top" :class="{'align-middle': item.encrypted}">
+                        <div class="w-full" v-if="settings['team:environment-variable-view'] || user.admin">
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/sections/Environment.vue:32]: I believe we'll still want the `item.encrypted` feature to show the padlock icon. The `settings['team:environment-variable-view']` can be added as an extra fork the if statement

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 3]
```diff
@@ -66,9 +68,8 @@
                         No Environment Variables Defined
                     </td>
                 </tr>
-                <!-- Empty row to differentiate between the existing env vars, and the input form row-->
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/sections/Environment.vue:69]: I think this comment is still providing useful explaination here!

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 4]
```diff
@@ -186,11 +184,6 @@ export default {
             this.editable.settings.env.splice(index, 1)
         }
     },
-    mounted () {
-        this.editable.settings.env.forEach(field => {
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/sections/Environment.vue:190]: Are you able to help me understand why this check no longer needs to run on mount?

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 5]
```diff
@@ -155,11 +157,7 @@ export default {
                         field.policy = false
                     }
                     if (field.policy === undefined && this.envVarNames[field.name] !== i) {
-                        field.error = 'Field has duplicate name'
```
Comment (Pezmc) [inline frontend/src/pages/admin/Template/sections/Environment.vue:158]: I'm not sure I follow the reason behind this change, why do we no longer need to check for duplicate ENV keys or empty names?

**Review by Pezmc — CHANGES_REQUESTED** [review]
Hi @sumitshinde-84, thank you for your contributions so far! I have a couple of open questions that I've dropped at comments on this PR.

Please note that as per our [definition of none](https://github.com/flowforge/website/blob/main/src/handbook/development/releases/planning.md#defining-done) and the PR check list, this PR will need test coverage before it can be merged. Happy to give a few pointers on that if you'd like. I'd suggest E2E tests using Cypress are likely most appropriate.

File: frontend/src/pages/admin/Template/sections/Environment.vue (hunk 2 above)
Comment (sumitshinde-84) [reply frontend/src/pages/admin/Template/sections/Environment.vue:32]: @Pezmc  i have tried  ```v-if="!item.encrypted"``` but its not hiding variable values, if i removed not(!) then its works but padlock icon is not visible .

File: frontend/src/pages/admin/Template/sections/Environment.vue [hunk 6]
```diff
@@ -116,6 +116,8 @@ export default {
         }
     },
     computed: {
+        ...mapState('account', ['team', 'settings']),
+        ...mapState('account', ['user', 'team', 'teams']),
```
[the remark this answers is not in the document]
Comment (sumitshinde-84) [reply frontend/src/pages/admin/Template/sections/Environment.vue]: Thank you so much, @Pezmc, for bringing this to my attention and providing such valuable suggestions. I greatly appreciate your input. I will commence writing the test for it promptly.

File: frontend/src/pages/admin/Template/sections/Environment.vue (hunk 2 above)
Comment (sumitshinde-84) [reply frontend/src/pages/admin/Template/sections/Environment.vue:32]: WhatsApp Image 2023-06-28 at 8 27 21 PM

done now pedlock icon is visible

**Comment by sumitshinde-84** [comment]
Hi @Pezmc, I've incorporated Cypress to test the UI changes. If you come across any mistakes, please don't hesitate to let me know. Your feedback is highly appreciated.

**Comment by MarianRaphael** [comment]
Hello @sumitshinde-84 Thank you for your valuable contribution! The feature you've implemented to allow administrators to control the view restrictions of Environment Variable values certainly adds a layer of privacy by limiting access to sensitive information.

However, after reviewing, I noticed a mismatch between the developed aspects and what was outlined in the original issue. The issue specifically mentioned the necessity for the Team Owner to have the ability to restrict the view on a variable-by-variable basis. Your implementation seems to provide control to administrators in general, rather than focusing on the specific role of the Team Owner, and doesn't seem to include the granularity required to manage access on a var by var basis.

Could you please take another look at this feature and adjust the implementation accordingly?
Thank you again for your hard work, and I look forward to seeing the updated feature.

**Pull request closed without merging**